File: programcreek/python/01/01-12-merge-sorted-arrays.py
def mergeArrays(A, B):

    aLen = len(A)
    bLen = len(B)

    for i in range(bLen):
        A.append(0)

    while aLen > 0 and bLen > 0:

        if A[aLen - 1] > B[bLen - 1]:
            A[aLen + bLen - 1] = A[aLen - 1]
            aLen -= 1
        else :
            A[aLen + bLen - 1] = B[bLen - 1]
            bLen -= 1

        print A, B

    while bLen > 0:
        A[aLen + bLen - 1] = B[bLen - 1]
        bLen -= 1

    return A





print mergeArrays([2,3,4], [4,5,6,7,8] )

File: programcreek/js/01/01-12-merge-sorted-arrays.js
function mergeArrays(A, B) {


    var m = A.length;
    var n = B.length;


    B.forEach(function() {A.push(0)});

    while (m > 0 && n > 0) {
        if (A[m - 1] > B[n - 1]) {
            A[m + n - 1] = A[m - 1];
            m --;
        }
        else {
            A[m + n - 1] = B[n - 1];
            n --;
        }
    }

    while (n > 0) {
        A[m + n - 1] = B[n - 1];
        n --;
    }

    return  A;
}



console.log( mergeArrays([2,3,4], [4,5,6,7,8] )  );